### ISSUE TITLE
ashell: LED0 does not work on Arduino 101

### ISSUE THREAD

**Issue by poussa** [01org/zephyr.js#693]
The LED0 (onboard) does not work with the ashell (master, d6f1ce185212e892c73dbacd0b865717600b09c8)

I tried with following samples 

samples/arduino/basics/Blink.js
samples/arduino/digital/Button.js
samples/TrafficLight.js

They do not work with shell but do work without it.

**Comment by poussa** [comment]
@brianjjones please take a look asap since we are using the shell during the next event.

**Comment by kenchris** [comment]
I had RGB working fine yesterday but after upgrading the GroveLCD example doesn't work for me. (both without ashell)

**Comment by nagineni** [comment]
I just tried with samples/ButtonLEDs.js from latest master and it is working fine.

**Comment by poussa** [comment]
@kenchris what do you mean by upgrading? From which commit to which?

**Comment by nagineni** [comment]
Issue is only with LED0. Onboard LED1 is working fine.

**Comment by poussa** [comment]
Yes, as the title says: with **ashell** **LED0** does **not** work. With ashell LED1 works. Without ashell I have not discovered any issues either.

**Comment by brianjjones** [comment]
Strange, I saw it working yesterday with ashell.  But I'll take a look this morning and confirm.

**Comment by poussa** [comment]
@brianjjones Any news on this one? The issue is still there with the latest master (249f119cf6f88da5a54d1a6669a59631e0092329)

Easy way to test is TrafficLights.js with ashell. The onboard led (green, pin13, LED0) stays off but the other 2 leds blink.

**Comment by poussa** [comment]
By the time I connect from the Browser to the device, I see on the console:

```
Hello, ZJS world!

src/zjs_callbacks.c:501 zjs_call_callback():
[ERROR] callback does not exist: 0
```

Don't know if that is relevant or not.

**Comment by brianjjones** [comment]
This is my priority today, I should have a fix soon.

**Comment by grgustaf** [comment]
I believe the problem here is that our "LED0" is tied to IO13, which in ashell is being used for SPI to talk to the flash filesystem. I'll confirm that by enabling SPI in a regular build and I expect that will break the sample.

If this is right, I don't think there is any way around it and we should probably rearrange our LED names so that LED0 becomes LED2 and has a note of the circumstances limiting its usage.

**Comment by grgustaf** [comment]
I did confirm that adding one line to the prj.conf: CONFIG_SPI=y, breaks this in the normal (non-ashell) build. So that is indeed the problem and I don't think there is a solution for ashell since it needs SPI for the filesystem.

**Comment by poussa** [comment]
Nothing we can do but document. Once that is done I'll close this bug.

**Comment by grgustaf** [comment]
Note to self: document in a101 pins, in ashell, and in future in filesystem/spi docs as well.

**Comment by grgustaf** [comment]
Fixed via documentation in #930.
